common/djangoapps/student/models.py

```diff
@@ -190,6 +190,16 @@ def user_by_anonymous_id(uid):
         return None
 
 
+def get_users_by_email(domain):
+    if domain is None:
+        return None
+
+    try:
+        return User.objects.filter(email__endswith=('@' + domain))
```

Comment:
Can you add an additional filter such that only users with is_active = 0 are returned.